lib/Transforms/Scalar/UnifyAssumes.cc

```diff
@@ -172,10 +172,15 @@ void UnifyAssumesPass::processAssertInst(CallInst &CI, AllocaInst &flag) {
   llvm::CallSite CS(&CI);
   IRBuilder<> B(bb);
   B.SetInsertPoint(&CI);
+  Value *conseq = CS.getArgument(0);
+  // skip transformation if verifier.assert(true)
+  if (auto *conseq_const = dyn_cast<llvm::Constant>(conseq)) {
+    if (conseq_const->isOneValue())
```

Comment:
in C 0 is false and non zero is true

you are assuming that 1 is true 

while this does not matter because the argument is i1 it is better to do it properly. I think there is also a method to check that a constant is false

Reply:
Will update by using `isZeroValue`.